Create an example to get metrics from ActiveMQ

wpjunior opened the pull request:
This example shows how to get Topic, Queue, General metrics via JMX attributes of activemq.

brian-brazil commented on example_configs/activemq.yml:
```diff
@@ -0,0 +1,48 @@
+hostPort:
```
You can remove the empty bits, they'll cause confusion

wpjunior replied:
Thanks removed =)

brian-brazil commented on example_configs/activemq.yml:
```diff
@@ -0,0 +1,48 @@
+blacklistObjectNames:
+  - "org.apache.activemq:clientId=*,*"
+whitelistObjectNames:
+  - "java.lang:type=OperatingSystem"
```
You don't need these, they're handled by default

brian-brazil commented on example_configs/activemq.yml:
```diff
@@ -0,0 +1,48 @@
+    destination: $2
+
+- pattern: org.apache.activemq<type=Broker, brokerName=(\S*)><>CurrentConnectionsCount
+  name: activemq_current_connections_count
```
Gauges shouldn't end in _count

wpjunior replied:
I renamed to just `activemq_current_connections`

wpjunior commented:
@brian-brazil this is a sample output:
```
# HELP activemq_topic_enqueue_count Number of messages that have been sent to the destination. (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>EnqueueCount)
# TYPE activemq_topic_enqueue_count untyped
activemq_topic_enqueue_count{broker="guest.com",destination="mytopic>",} 29352.0
# HELP activemq_topic_max_audit_depth Max audit depth (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>MaxAuditDepth)
# TYPE activemq_topic_max_audit_depth untyped
activemq_topic_max_audit_depth{broker="guest.com",destination="mytopic>",} 10000.0
# HELP activemq_queue_average_enqueue_time Average time a message was held on this destination. (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>AverageEnqueueTime)
# TYPE activemq_queue_average_enqueue_time untyped
activemq_queue_average_enqueue_time{broker="guest.com",destination="myqueue>",} 72.40336634505037
# HELP activemq_queue_expired_count Number of messages that have been expired. (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>ExpiredCount)
# TYPE activemq_queue_expired_count untyped
activemq_queue_expired_count{broker="guest.com",destination="myqueue>",} 7.0
# HELP activemq_topic_max_message_size Max message size on this destination (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>MaxMessageSize)
# TYPE activemq_topic_max_message_size untyped
activemq_topic_max_message_size{broker="guest.com",destination="mytopic>",} 359899.0
# HELP activemq_topic_producer_count Number of producers attached to this destination (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>ProducerCount)
# TYPE activemq_topic_producer_count untyped
activemq_topic_producer_count{broker="guest.com",destination="mytopic>",} 0.0
# HELP activemq_queue_max_page_size Maximum number of messages to be paged in (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>MaxPageSize)
# TYPE activemq_queue_max_page_size untyped
activemq_queue_max_page_size{broker="guest.com",destination="myqueue>",} 200.0
# HELP activemq_topic_max_enqueue_time The longest time a message was held on this destination (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>MaxEnqueueTime)
# TYPE activemq_topic_max_enqueue_time untyped
activemq_topic_max_enqueue_time{broker="guest.com",destination="mytopic>",} 24.0
# HELP activemq_queue_average_message_size Average message size on this destination (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>AverageMessageSize)
# TYPE activemq_queue_average_message_size untyped
activemq_queue_average_message_size{broker="guest.com",destination="myqueue>",} 33276.0
# HELP activemq_topic_blocked_sends Number of messages blocked for flow control (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>BlockedSends)
# TYPE activemq_topic_blocked_sends untyped
activemq_topic_blocked_sends{broker="guest.com",destination="mytopic>",} 0.0
# HELP activemq_queue_producer_count Number of producers attached to this destination (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>ProducerCount)
# TYPE activemq_queue_producer_count untyped
activemq_queue_producer_count{broker="guest.com",destination="myqueue>",} 0.0
# HELP activemq_queue_blocked_producer_warning_interval Blocked Producer Warning Interval (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>BlockedProducerWarningInterval)
# TYPE activemq_queue_blocked_producer_warning_interval untyped
activemq_queue_blocked_producer_warning_interval{broker="guest.com",destination="myqueue>",} 30000.0
# HELP activemq_queue_blocked_sends Number of messages blocked for flow control (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>BlockedSends)
# TYPE activemq_queue_blocked_sends untyped
activemq_queue_blocked_sends{broker="guest.com",destination="myqueue>",} 0.0
# HELP activemq_topic_memory_usage_byte_count Memory used by undelivered messages in bytes (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>MemoryUsageByteCount)
# TYPE activemq_topic_memory_usage_byte_count untyped
activemq_topic_memory_usage_byte_count{broker="guest.com",destination="mytopic>",} 0.0
# HELP activemq_topic_dequeue_count Number of messages that has been acknowledged (and removed) from the destination. (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>DequeueCount)
# TYPE activemq_topic_dequeue_count untyped
activemq_topic_dequeue_count{broker="guest.com",destination="mytopic>",} 38.0
# HELP activemq_topic_temp_usage_percent_usage The percentage of the temp usage limit used (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>TempUsagePercentUsage)
# TYPE activemq_topic_temp_usage_percent_usage untyped
activemq_topic_temp_usage_percent_usage{broker="guest.com",destination="mytopic>",} 0.0
# HELP activemq_topic_in_flight_count Number of messages that have been dispatched to, but not acknowledged by, consumers. (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>InFlightCount)
# TYPE activemq_topic_in_flight_count untyped
activemq_topic_in_flight_count{broker="guest.com",destination="mytopic>",} 0.0
# HELP activemq_topic_use_cache Caching is allowed (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>UseCache)
# TYPE activemq_topic_use_cache untyped
activemq_topic_use_cache{broker="guest.com",destination="mytopic>",} 1.0
# HELP activemq_queue_cursor_full Message cusor has reached its memory limit for paged in messages (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>CursorFull)
# TYPE activemq_queue_cursor_full untyped
activemq_queue_cursor_full{broker="guest.com",destination="myqueue>",} 0.0
# HELP activemq_queue_max_producers_to_audit Maximum number of producers to audit (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>MaxProducersToAudit)
# TYPE activemq_queue_max_producers_to_audit untyped
activemq_queue_max_producers_to_audit{broker="guest.com",destination="myqueue>",} 64.0
# HELP activemq_queue_prioritized_messages Prioritized messages is enabled (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>PrioritizedMessages)
# TYPE activemq_queue_prioritized_messages untyped
activemq_queue_prioritized_messages{broker="guest.com",destination="myqueue>",} 0.0
# HELP activemq_queue_dispatch_count Number of messages that has been delivered to consumers, including those not acknowledged (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>DispatchCount)
# TYPE activemq_queue_dispatch_count untyped
activemq_queue_dispatch_count{broker="guest.com",destination="myqueue>",} 23527.0
# HELP activemq_topic_consumer_count Number of consumers subscribed to this destination. (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>ConsumerCount)
# TYPE activemq_topic_consumer_count untyped
activemq_topic_consumer_count{broker="guest.com",destination="mytopic>",} 2.0
# HELP activemq_topic_always_retroactive Always treat consumers as retroactive (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>AlwaysRetroactive)
# TYPE activemq_topic_always_retroactive untyped
activemq_topic_always_retroactive{broker="guest.com",destination="mytopic>",} 0.0
# HELP activemq_topic_temp_usage_limit Temp usage limit, in bytes, assigned to this destination. (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>TempUsageLimit)
# TYPE activemq_topic_temp_usage_limit untyped
activemq_topic_temp_usage_limit{broker="guest.com",destination="mytopic>",} 5.36870912E8
# HELP activemq_topic_total_blocked_time Total time (ms) messages have been blocked by flow control (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>TotalBlockedTime)
# TYPE activemq_topic_total_blocked_time untyped
activemq_topic_total_blocked_time{broker="guest.com",destination="mytopic>",} 0.0
# HELP activemq_queue_always_retroactive Always treat consumers as retroactive (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>AlwaysRetroactive)
# TYPE activemq_queue_always_retroactive untyped
activemq_queue_always_retroactive{broker="guest.com",destination="myqueue>",} 0.0
# HELP activemq_connections_count Attribute exposed for management (org.apache.activemq<type=Broker, brokerName=guest.com><>TotalConnectionsCount)
# TYPE activemq_connections_count counter
# HELP activemq_queue_in_flight_count Number of messages that have been dispatched to, but not acknowledged by, consumers. (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>InFlightCount)
# TYPE activemq_queue_in_flight_count untyped
activemq_queue_in_flight_count{broker="guest.com",destination="myqueue>",} 0.0
# HELP activemq_topic_memory_usage_portion Portion of memory from the broker memory limit for this destination (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>MemoryUsagePortion)
# TYPE activemq_topic_memory_usage_portion untyped
activemq_topic_memory_usage_portion{broker="guest.com",destination="mytopic>",} 1.0
# HELP activemq_queue_cursor_memory_usage Message cursor memory usage, in bytes. (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>CursorMemoryUsage)
# TYPE activemq_queue_cursor_memory_usage untyped
activemq_queue_cursor_memory_usage{broker="guest.com",destination="myqueue>",} 0.0
# HELP activemq_queue_max_message_size Max message size on this destination (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>MaxMessageSize)
# TYPE activemq_queue_max_message_size untyped
activemq_queue_max_message_size{broker="guest.com",destination="myqueue>",} 359899.0
# HELP activemq_queue_min_enqueue_time The shortest time a message was held on this destination (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>MinEnqueueTime)
# TYPE activemq_queue_min_enqueue_time untyped
activemq_queue_min_enqueue_time{broker="guest.com",destination="myqueue>",} 4.0
# HELP activemq_queue_memory_usage_portion Portion of memory from the broker memory limit for this destination (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>MemoryUsagePortion)
# TYPE activemq_queue_memory_usage_portion untyped
activemq_queue_memory_usage_portion{broker="guest.com",destination="myqueue>",} 0.0
# HELP activemq_topic_max_page_size Maximum number of messages to be paged in (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>MaxPageSize)
# TYPE activemq_topic_max_page_size untyped
activemq_topic_max_page_size{broker="guest.com",destination="mytopic>",} 200.0
# HELP activemq_topic_queue_size Number of messages on this destination, including any that have been dispatched but not acknowledged (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>QueueSize)
# TYPE activemq_topic_queue_size untyped
activemq_topic_queue_size{broker="guest.com",destination="mytopic>",} 0.0
# HELP activemq_topic_expired_count Number of messages that have been expired. (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>ExpiredCount)
# TYPE activemq_topic_expired_count untyped
activemq_topic_expired_count{broker="guest.com",destination="mytopic>",} 0.0
# HELP activemq_queue_store_message_size The memory size of all messages in this destination's store. (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>StoreMessageSize)
# TYPE activemq_queue_store_message_size untyped
activemq_queue_store_message_size{broker="guest.com",destination="myqueue>",} 0.0
# HELP activemq_topic_average_blocked_time Average time (ms) messages have been blocked by flow control (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>AverageBlockedTime)
# TYPE activemq_topic_average_blocked_time untyped
activemq_topic_average_blocked_time{broker="guest.com",destination="mytopic>",} 0.0
# HELP activemq_queue_min_message_size Min message size on this destination (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>MinMessageSize)
# TYPE activemq_queue_min_message_size untyped
activemq_queue_min_message_size{broker="guest.com",destination="myqueue>",} 5476.0
# HELP activemq_topic_blocked_producer_warning_interval Blocked Producer Warning Interval (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>BlockedProducerWarningInterval)
# TYPE activemq_topic_blocked_producer_warning_interval untyped
activemq_topic_blocked_producer_warning_interval{broker="guest.com",destination="mytopic>",} 30000.0
# HELP activemq_queue_max_enqueue_time The longest time a message was held on this destination (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>MaxEnqueueTime)
# TYPE activemq_queue_max_enqueue_time untyped
activemq_queue_max_enqueue_time{broker="guest.com",destination="myqueue>",} 46343.0
# HELP activemq_current_connections_count Attribute exposed for management (org.apache.activemq<type=Broker, brokerName=guest.com><>CurrentConnectionsCount)
# TYPE activemq_current_connections_count gauge
# HELP activemq_queue_temp_usage_percent_usage The percentage of the temp usage limit used (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>TempUsagePercentUsage)
# TYPE activemq_queue_temp_usage_percent_usage untyped
activemq_queue_temp_usage_percent_usage{broker="guest.com",destination="myqueue>",} 0.0
# HELP activemq_queue_memory_limit Memory limit, in bytes, used by undelivered messages before paging to temporary storage. (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>MemoryLimit)
# TYPE activemq_queue_memory_limit untyped
activemq_queue_memory_limit{broker="guest.com",destination="myqueue>",} 8388608.0
# HELP activemq_topic_prioritized_messages Prioritized messages is enabled (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>PrioritizedMessages)
# TYPE activemq_topic_prioritized_messages untyped
activemq_topic_prioritized_messages{broker="guest.com",destination="mytopic>",} 0.0
# HELP activemq_topic_average_enqueue_time Average time a message was held on this destination. (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>AverageEnqueueTime)
# TYPE activemq_topic_average_enqueue_time untyped
activemq_topic_average_enqueue_time{broker="guest.com",destination="mytopic>",} 17.55263157894737
# HELP activemq_topic_memory_limit Memory limit, in bytes, used by undelivered messages before paging to temporary storage. (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>MemoryLimit)
# TYPE activemq_topic_memory_limit untyped
activemq_topic_memory_limit{broker="guest.com",destination="mytopic>",} 5.36870912E8
# HELP activemq_queue_paused Dispatch to consumers is paused (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>Paused)
# TYPE activemq_queue_paused untyped
activemq_queue_paused{broker="guest.com",destination="myqueue>",} 0.0
# HELP activemq_topic_producer_flow_control Flow control is enabled for producers (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>ProducerFlowControl)
# TYPE activemq_topic_producer_flow_control untyped
activemq_topic_producer_flow_control{broker="guest.com",destination="mytopic>",} 1.0
# HELP activemq_topic_min_enqueue_time The shortest time a message was held on this destination (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>MinEnqueueTime)
# TYPE activemq_topic_min_enqueue_time untyped
activemq_topic_min_enqueue_time{broker="guest.com",destination="mytopic>",} 14.0
# HELP activemq_queue_dequeue_count Number of messages that has been acknowledged (and removed) from the destination. (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>DequeueCount)
# TYPE activemq_queue_dequeue_count untyped
activemq_queue_dequeue_count{broker="guest.com",destination="myqueue>",} 23534.0
# HELP activemq_queue_producer_flow_control Flow control is enabled for producers (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>ProducerFlowControl)
# TYPE activemq_queue_producer_flow_control untyped
activemq_queue_producer_flow_control{broker="guest.com",destination="myqueue>",} 1.0
# HELP activemq_queue_cursor_percent_usage Percentage of memory limit used (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>CursorPercentUsage)
# TYPE activemq_queue_cursor_percent_usage untyped
activemq_queue_cursor_percent_usage{broker="guest.com",destination="myqueue>",} 0.0
# HELP activemq_topic_memory_percent_usage The percentage of the memory limit used (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>MemoryPercentUsage)
# TYPE activemq_topic_memory_percent_usage untyped
activemq_topic_memory_percent_usage{broker="guest.com",destination="mytopic>",} 0.0
# HELP activemq_topic_dispatch_count Number of messages that has been delivered to consumers, including those not acknowledged (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>DispatchCount)
# TYPE activemq_topic_dispatch_count untyped
activemq_topic_dispatch_count{broker="guest.com",destination="mytopic>",} 38.0
# HELP activemq_queue_temp_usage_limit Temp usage limit, in bytes, assigned to this destination. (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>TempUsageLimit)
# TYPE activemq_queue_temp_usage_limit untyped
activemq_queue_temp_usage_limit{broker="guest.com",destination="myqueue>",} 5.36870912E8
# HELP activemq_queue_cache_enabled Caching is enabled (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>CacheEnabled)
# TYPE activemq_queue_cache_enabled untyped
activemq_queue_cache_enabled{broker="guest.com",destination="myqueue>",} 1.0
# HELP activemq_queue_use_cache Caching is allowed (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>UseCache)
# TYPE activemq_queue_use_cache untyped
activemq_queue_use_cache{broker="guest.com",destination="myqueue>",} 1.0
# HELP activemq_queue_memory_usage_byte_count Memory used by undelivered messages in bytes (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>MemoryUsageByteCount)
# TYPE activemq_queue_memory_usage_byte_count untyped
activemq_queue_memory_usage_byte_count{broker="guest.com",destination="myqueue>",} 0.0
# HELP activemq_queue_total_blocked_time Total time (ms) messages have been blocked by flow control (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>TotalBlockedTime)
# TYPE activemq_queue_total_blocked_time untyped
activemq_queue_total_blocked_time{broker="guest.com",destination="myqueue>",} 0.0
# HELP activemq_queue_enqueue_count Number of messages that have been sent to the destination. (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>EnqueueCount)
# TYPE activemq_queue_enqueue_count untyped
activemq_queue_enqueue_count{broker="guest.com",destination="myqueue>",} 23527.0
# HELP activemq_queue_queue_size Number of messages on this destination, including any that have been dispatched but not acknowledged (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>QueueSize)
# TYPE activemq_queue_queue_size untyped
activemq_queue_queue_size{broker="guest.com",destination="myqueue>",} 0.0
# HELP activemq_queue_dlq Dead Letter Queue (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>DLQ)
# TYPE activemq_queue_dlq untyped
activemq_queue_dlq{broker="guest.com",destination="myqueue>",} 0.0
# HELP activemq_topic_min_message_size Min message size on this destination (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>MinMessageSize)
# TYPE activemq_topic_min_message_size untyped
activemq_topic_min_message_size{broker="guest.com",destination="mytopic>",} 5230.0
# HELP activemq_topic_store_message_size The memory size of all messages in this destination's store. (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>StoreMessageSize)
# TYPE activemq_topic_store_message_size untyped
activemq_topic_store_message_size{broker="guest.com",destination="mytopic>",} 0.0
# HELP activemq_queue_average_blocked_time Average time (ms) messages have been blocked by flow control (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>AverageBlockedTime)
# TYPE activemq_queue_average_blocked_time untyped
activemq_queue_average_blocked_time{broker="guest.com",destination="myqueue>",} 0.0
# HELP activemq_topic_average_message_size Average message size on this destination (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>AverageMessageSize)
# TYPE activemq_topic_average_message_size untyped
activemq_topic_average_message_size{broker="guest.com",destination="mytopic>",} 28943.0
# HELP activemq_queue_consumer_count Number of consumers subscribed to this destination. (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>ConsumerCount)
# TYPE activemq_queue_consumer_count untyped
activemq_queue_consumer_count{broker="guest.com",destination="myqueue>",} 1.0
# HELP activemq_queue_memory_percent_usage The percentage of the memory limit used (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>MemoryPercentUsage)
# TYPE activemq_queue_memory_percent_usage untyped
activemq_queue_memory_percent_usage{broker="guest.com",destination="myqueue>",} 0.0
# HELP activemq_topic_forward_count Number of messages that have been forwarded (to a networked broker) from the destination. (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>ForwardCount)
# TYPE activemq_topic_forward_count untyped
activemq_topic_forward_count{broker="guest.com",destination="mytopic>",} 0.0
# HELP activemq_queue_max_audit_depth Max audit depth (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>MaxAuditDepth)
# TYPE activemq_queue_max_audit_depth untyped
activemq_queue_max_audit_depth{broker="guest.com",destination="myqueue>",} 10000.0
# HELP activemq_topic_dlq Dead Letter Queue (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>DLQ)
# TYPE activemq_topic_dlq untyped
activemq_topic_dlq{broker="guest.com",destination="mytopic>",} 0.0
# HELP activemq_queue_forward_count Number of messages that have been forwarded (to a networked broker) from the destination. (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Queue, destinationName=myqueue><>ForwardCount)
# TYPE activemq_queue_forward_count untyped
activemq_queue_forward_count{broker="guest.com",destination="myqueue>",} 0.0
# HELP activemq_topic_max_producers_to_audit Maximum number of producers to audit (org.apache.activemq<type=Broker, brokerName=guest.com, destinationType=Topic, destinationName=mytopic><>MaxProducersToAudit)
# TYPE activemq_topic_max_producers_to_audit untyped
activemq_topic_max_producers_to_audit{broker="guest.com",destination="mytopic>",} 64.0
# HELP jmx_scrape_duration_seconds Time this JMX scrape took, in seconds.
# TYPE jmx_scrape_duration_seconds gauge
# HELP jmx_scrape_error Non-zero if this scrape failed.
# TYPE jmx_scrape_error gauge
# HELP jmx_exporter_build_info A metric with a constant '1' value labeled with the version of the JMX exporter.
# TYPE jmx_exporter_build_info gauge
# HELP jmx_config_reload_success_total Number of times configuration have successfully been reloaded.
# TYPE jmx_config_reload_success_total counter
# HELP jmx_config_reload_failure_total Number of times configuration have failed to be reloaded.
# TYPE jmx_config_reload_failure_total counter
```

brian-brazil commented on example_configs/activemq.yml:
```diff
@@ -0,0 +1,35 @@
+  type: GAUGE
+
+- pattern: org.apache.activemq<type=Broker, brokerName=(\S*)><>TotalConnectionsCount
+  name: activemq_connections_count
```
Counters should end in _total

brian-brazil commented on example_configs/activemq.yml:
```diff
@@ -0,0 +1,35 @@
+  name: activemq_queue_$3
+  attrNameSnakeCase: true
+  labels:
+    broker: $1
```
Broker looks to be a target label, remove it.

brian-brazil commented on example_configs/activemq.yml:
```diff
@@ -0,0 +1,35 @@
+    destination: $2
+
+- pattern: org.apache.activemq<type=Broker, brokerName=(\S*)><>CurrentConnectionsCount
+  name: activemq_current_connections
```
activemq_connections would be more common

wpjunior commented:
Hi @brian-brazil I've added new global metrics: `activemq_consumers_total, activemq_dequeue_total, activemq_enqueue_total and activemq_messages_total, activemq_store_usage_ratio, activemq_temp_usage_ratio and activemq_memory_usage_ratio`

brian-brazil commented on example_configs/activemq.yml:
```diff
@@ -0,0 +1,60 @@
+  name: activemq_connections
+  type: GAUGE
+
+- pattern: org.apache.activemq<type=Broker, brokerName=(\S*)><>TotalConnectionsCount
```
You should be able to combine all these Total.*Count rules into one.

wpjunior replied:
Done

brian-brazil commented on example_configs/activemq.yml:
```diff
@@ -0,0 +1,60 @@
+  name: activemq_messages_total
+  type: COUNTER
+
+- pattern: org.apache.activemq<type=Broker, brokerName=(\S*)><>TempPercentUsage
```
Is the raw data for this available?

wpjunior replied:
Yes I have from `jconsole`: 
![image](https://user-images.githubusercontent.com/691587/64175823-8b2a4280-ce32-11e9-9ffb-0d95ba24ce3c.png)

brian-brazil replied:
That's a no then, there's only the percentage.

brian-brazil commented:
Thanks!
